### PR TITLE
chore: remove the folder 'logs'  in root and replace baidu.com with github.com in test cases

### LINKED ISSUE #2334: test: replace `baidu.com` to another domain in test cases
use `baidu.com` is unstable in Github Action. 

maybe we can use `www.apple.com` for testing.

### LINKED ISSUE #2370: chore: remove folder `logs` in root
We can delete the "logs" folder from the "root" directory.

https://github.com/apache/apisix/tree/master/logs

### PR DESCRIPTION
**Pull request by dabue**
fix #2370 to remove the logs dir.
fix #2334 to replace baidu.com with github.com in test cases

### CODE REVIEW

**Comment by membphis** [comment]
@dabue CI failed, please take a look at it.

https://github.com/apache/apisix/pull/2374/checks?check_run_id=1234496186#step:6:51

**Comment by dabue** [comment]
I cleared the folder logs and not removed it.  It is required when the apisix is running.

**Comment by membphis** [comment]
> I cleared the folder logs and not removed it. It is required when the apisix is running.

we can create the `logs` folder in Makefile or `bin/apisix` before apisix is starting.

**Comment by dabue** [comment]
> > I cleared the folder logs and not removed it. It is required when the apisix is running.
> 
> we can create the `logs` folder in Makefile or `bin/apisix` before apisix is starting.

ok, I'll update it.

**Review by membphis — APPROVED** [review]

**Comment by Yiyiyimu** [comment]
Maybe we could also update baidu.com to github.com in doc in this PR:
https://github.com/apache/apisix/blob/master/doc/plugins/wolf-rbac.md
https://github.com/apache/apisix/blob/master/doc/zh-cn/plugins/wolf-rbac.md

**Comment by membphis** [comment]
> Maybe we could also update baidu.com to github.com in doc in this PR:
> https://github.com/apache/apisix/blob/master/doc/plugins/wolf-rbac.md
> https://github.com/apache/apisix/blob/master/doc/zh-cn/plugins/wolf-rbac.md

fix them in a new PR, it is fine

**Review by Yiyiyimu — APPROVED** [review]

File: bin/apisix [hunk 1]
```diff
@@ -1065,6 +1065,9 @@ local openresty_args = [[openresty  -p ]] .. apisix_home .. [[ -c ]]
                        .. apisix_home .. [[/conf/nginx.conf]]
 
 function _M.start(...)
+
+    local cmd_logs = "mkdir -p " ..  apisix_home .. "/logs"
```
Comment (moonming) [inline bin/apisix:1069]: I don't think is ​​a good idea to create directories.  What if the user start apisix by sudo?  Does the worker still have permission to write logs?

Comment (moonming) [reply bin/apisix:1069]: @dabue @membphis

Comment (dabue) [reply bin/apisix:1069]: > @dabue @membphis

The directory logs is required when the APISIX is running. On the first running, it needs to be created if we keep the logs removed from the root, and should have the appropriate permissions.
when the directory already exists, whether the directory is created by the first running or is from the beginning,  the permission to write logs are same.
I'll do a test to verify it.

**Pull request merged**